server/controllers/alertsController.ts

```diff
@@ -34,8 +36,14 @@ export default class AlertsController {
     // Get prisoner data for banner and for use in alerts generation
     const prisonerData = await this.prisonerSearchService.getPrisonerDetails(req.params.prisonerNumber)
 
+    // Set role based permissions
+    const { authorities: roles = [] } = jwtDecode(res.locals.user.token) as { authorities?: string[] }
+    const canUpdateAlert =
+      roles.some(role => role === Role.UpdateAlert) &&
+      (res.locals.user.activeCaseLoadId === prisonerData.prisonId || ['OUT', 'TRN'].includes(prisonerData.prisonId))
```

Comment:
I wonder if these three lines could be made into a function that takes a role as an argument and returns true/false

Reply:
I thought about that, but the next tickets on Case Notes use slightly different logic, so there'd be no reuse (at least for now). If a few more tickets use the exact logic, then would be worth doing.

Reply:
Ah cool - yeah lets see how the rest shake out at least